src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -1433,6 +1413,183 @@ internal virtual HttpResponseMessage GetResponse(HttpClient client, HttpRequestM
+            }
+            catch (Exception ex)
+            {
+                // Just in case there are any edge cases we missed, we don't break workflows with an exception
+                WriteVerbose($"Failed to Write WebRequest Verbose Info: {ex} {ex.StackTrace}");
+            }
+        }
+
+        private void WriteWebRequestDebugInfo(HttpRequestMessage request)
+        {
+            try
+            {
+                StringBuilder debugBuilder = new("WebRequest Detail");
+
+                if (!string.IsNullOrEmpty(request.RequestUri.Query))
+                {
+                    debugBuilder.AppendLine(DebugHeaderPrefix + "QUERY");
+                    string[] queryParams = request.RequestUri.Query.TrimStart('?').Split('&');
+                    foreach (string param in queryParams)
+                    {
+                        debugBuilder.AppendLine(param);
+                    }
+                }
+
+                debugBuilder.AppendLine(DebugHeaderPrefix + "HEADERS");
+
+                List<KeyValuePair<string, IEnumerable<string>>> allHeaders = new();
+                foreach (HttpHeaders headerSource in new HttpHeaders[]
+                {
+                request.Headers,
+                request.Content?.Headers,
+                })
+                {
+                    if (headerSource is not null)
+                    {
+                        allHeaders.AddRange(headerSource);
+                    }
+                }
+                foreach (var header in allHeaders.OrderBy(h => h.Key))
+                {
+                    debugBuilder.AppendLine($"{header.Key}: {string.Join(", ", header.Value)}");
+                }
+
+                if (request.Content is not null)
+                {
+                    debugBuilder
+                    .AppendLine(DebugHeaderPrefix + "BODY")
+                    .AppendLine(request.Content switch
+                    {
+                        StringContent stringContent => stringContent
+                        .ReadAsStringAsync(_cancelToken.Token)
+                        .GetAwaiter().GetResult(),
+                        MultipartFormDataContent multipartContent => "=> Multipart Form Content"
+                        + Environment.NewLine
+                        + multipartContent.ReadAsStringAsync(_cancelToken.Token)
+                        .GetAwaiter().GetResult(),
+                        ByteArrayContent byteContent => InFile is not null
+                        ? "[Binary content: "
+                        + GetFriendlyContentLength(byteContent.Headers.ContentLength)
+                        + "]"
+                    : byteContent.ReadAsStringAsync(_cancelToken.Token).GetAwaiter().GetResult(),
+                    StreamContent streamContent => "[Stream content: " + GetFriendlyContentLength(streamContent.Headers.ContentLength) + "]",
+                    _ => "[Unknown content type]",
+                });
```

Comment:
It looks terrible. Could you have done a better code formatting?

Reply:
OK I don't know what happened here, it didn't look terrible before, the indentations were previously present, will fix.